docs/diagrams/AddBooleanCommandObjectDiagram.puml

```diff
@@ -0,0 +1,42 @@
+@startuml
+package Gates {
```

Comment:
Any reason you name them States? Seems harder to deal with than just using their names.